CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -611,14 +589,10 @@ public function emailReceipt(&$params) {
     // assign trxn details
     $this->assign('trxn_id', CRM_Utils_Array::value('trxn_id', $params));
     $this->assign('receive_date', CRM_Utils_Array::value('trxn_date', $params));
-    $paymentInstrument = CRM_Contribute_PseudoConstant::paymentInstrument();
-    if (array_key_exists('payment_instrument_id', $params)) {
-      $this->assign('paidBy',
-        CRM_Utils_Array::value($params['payment_instrument_id'],
-          $paymentInstrument
-        )
-      );
-    }
+    $paymentInstruments = CRM_Contribute_PseudoConstant::paymentInstrument();
+    $this->assign('paidBy',
+      CRM_Utils_Array::value($params['payment_instrument_id'], $paymentInstruments)
```

Comment:
$this->assign('paidBy', CRM_Core_PseudoConstant::getLabel('CRM_Contribute_BAO_Contribute', 'payment_instrument_id, $params['payment_instrument_id']); 

Is the way we are going on these